app/RecordCollectors/OpenCorporates.php

```diff
@@ -0,0 +1,217 @@
+		],
+		'Vendors' => [
+			'companyNumber' => 'registration_number_1',
+			'taxNumber' => 'registration_number_2'
+		],
+	];
+
+	/** {@inheritdoc} */
+	public $formFieldsToRecordMap = [
+		'Accounts' => [
+			'resultsCompanyName' => 'accountname',
+			'resultsCompanyCompany_number' => 'registration_number_1',
+			'resultsCompanyRegistry_url' => 'description'
+		],
+		'Leads' => [
+			'resultsCompanyName' => 'company',
+			'resultsCompanyCompany_number' => 'registration_number_1',
+			'resultsCompanyRegistry_url' => 'description'
+		],
+		'Vendors' => [
+			'resultsCompanyName' => 'vendorname',
+			'resultsCompanyCompany_number' => 'registration_number_1',
+			'resultsCompanyRegistry_url' => 'description'
+		],
+		'Partners' => [
+			'resultsCompanyName' => 'subject',
+			'resultsCompanyRegistry_url' => 'description'
+		],
+		'Competition' => [
+			'resultsCompanyName' => 'subject',
+			'resultsCompanyRegistry_url' => 'description'
+		],
+	];
+
+	/** @var string OpenCorporates sever address */
+	private $url = 'https://api.opencorporates.com/';
+
+	/** @var string API version. */
+	const API_VERSION = 'v0.4';
+
+	/** {@inheritdoc} */
+	public function search(): array
+	{
+		$countryCode = $this->request->getByType('countryCode', 'Text');
+		$companyNumber = str_replace([' ', ',', '.', '-'], '', $this->request->getByType('companyNumber', 'Text'));
+		$taxNumber = str_replace([' ', ',', '.', '-'], '', $this->request->getByType('taxNumber', 'Text'));
+		if (!$this->isActive() && empty($countryCode) && (empty($companyNumber) || empty($taxNumber))) {
+			return [];
+		}
+
+		$params = [];
+		if (!empty($companyNumber)) {
+			$params['companyNumber'] = $companyNumber;
+		}
+		if (!empty($taxNumber)) {
+			$params['taxNumber'] = $taxNumber;
+		}
+		$this->getDataFromApi($countryCode, $params);
+		$this->loadData();
+		return $this->response;
+	}
+
+	/**
+	 * Function fetching company data by params.
+	 *
+	 * @param string   $countryCode
+	 * @param string[] $params
+	 *
+	 * @return void
+	 */
+	private function getDataFromApi(string $countryCode, array $params): void
+	{
+		$response = [];
+		if (isset($params['companyNumber'])) {
+			$response = $this->apiConnection($countryCode, $params['companyNumber']);
+		}
+		if (isset($params['taxNumber']) && !isset($response['errorCode'])) {
+			$response = $this->apiConnection($countryCode, $params['taxNumber']);
+		}
+
+		$this->data = $response;
+	}
+
+	/**
+	 * Function fetching and setting codes (countries, districts) for&from OpenCorporates.
+	 *
+	 * @param array $data
+	 *
+	 * @return void
+	 */
+	public function getCodesFromApi(array $data)
+	{
+		try {
+			$response = \App\Json::decode(\App\RequestHttp::getClient()->get($this->url . self::API_VERSION . '/jurisdictions')->getBody()->getContents());
+		} catch (\GuzzleHttp\Exception\ClientException $e) {
+			\App\Log::warning($e->getMessage(), 'RecordCollectors');
+			$this->response['error'] = $e->getMessage();
```

Comment:
if ($e->getCode() > 400) {
				$this->response['error'] = $e->getResponse()->getReasonPhrase();
			} else {
				$this->response['error'] = $e->getMessage();
			}